src/basic/log.h

```diff
@@ -32,6 +32,11 @@ typedef enum LogTarget{
 #define IS_SYNTHETIC_ERRNO(val)             ((val) >> 30 & 1)
 #define ERRNO_VALUE(val)                    (abs(val) & 255)
 
+/* The callback function to be invoked when syntax warnings are seen
+ * in the unit files. */
+typedef int (*log_syntax_callback_t)(const char *unit, int level, void *log_syntax_callback_userdata);
```

Comment:
why log_syntax specific? any error should be relevant I'd assume. Please make this generic.

If you really want to distinguish which way the erro message came in, then define a bunch of more flags fields tha are implicitly ORed into the error field or so, the way we OR in the synthetic bit already.

Reply:
What we are looking for here is syntax errors when verifying units - and having the unit name available is a must for the behavior that @keszybz asked for. Also, we don't really want to call this on any error, it doesn't make sense to call it eg. because by default we'll look for /etc/systemd/system but that will raise an ENOENT on images with usr-only, and so on.

In other words, there is no use case where this would be used besides syntax error checking.

Reply:
well, but i think log.c infra should be generic. if your callback then ignores half the params, that's fine, but I still think the infra should be generic.

Reply:
The unit name is not passed down as a parameter, so not sure how it could work if it moved to be called from a different log function? Besides, log.c already has a non-generic `log_syntax()` set of interfaces exposed which are separate APIs from the rest and for good reasons, so why would it be an issue to have a log_syntax-specific callback? I mean if `log_syntax()` didn't exist in the first place then sure, but it does. If in the future there are different needs we can revise, but at the moment there aren't, and to me it seems better to keep it simpler, shorter and to the point, no?